### PR TITLE
Init commit and solution of HW 7

### PR DESCRIPTION
**Pull request by Yury3077**

### CODE REVIEW

File: homework7/hw1.py [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+    },
+    "fourth": "RED",
+    "set_some": ("RED", "1", None, 6, True, (1, "RED")),
+}
+
+
+def find_occurrences(tree: dict, element: Any) -> int:
+    """
+    Find amount of element in dict(tree)
+    param:
```
Comment (AlekseiAfanasev) [inline homework7/hw1.py:34]: there is no need in params in docstrings if you have verbose representative names and typings

File: homework7/hw1.py [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+        tree (dict): dict that have different structures inside: str, list, tuple, dict, set, int, bool
+        element (any): value that we need to find in a dict
+    return: the number of occurrences of this element in the tree
+    """
+    count = 0
+    if isinstance(tree, dict):
+        tree = tree.values()
```
Comment (AlekseiAfanasev) [inline homework7/hw1.py:41]: this way you'll ignore occurences in keys

File: homework7/hw1.py [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+    for item in tree:
+        if item == element:
+            count += 1
+        if isinstance(item, dict):
+            elem_tree = item.values()
+            count += find_occurrences(elem_tree, element)
+        if isinstance(item, list) or isinstance(item, set) or isinstance(item, tuple):
```
Comment (AlekseiAfanasev) [inline homework7/hw1.py:48]: this way you'll ignore any custom iterable types. try to think about another check for iterables

File: homework7/hw2.py [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+def backspace_compare(first: str, second: str) -> bool:
+    """
+    Function that compare two str with symbol "#" means a backspace
+    param:
+        first (str): input string
+        second(str) : input string
+    return: True is strings are equal and False if not
+    """
+    if str_without_backspace(first) == str_without_backspace(second):
+        return True
```
Comment (AlekseiAfanasev) [inline homework7/hw2.py:32]: when you do conditions with simple "return True"/"return False" in condition branches you may replace by by simple "return {condition}"

File: homework7/hw2.py [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+    """
+    Find symbol "#" in str and delete previous letter in string
+    param: some_str (str): input string
+    return: String without "#"
+    """
+    str_without_backspace_1 = ""
+    for symbol in some_str:
+        if symbol == "#":
+            str_without_backspace_1 = str_without_backspace_1[
+                0 : len(str_without_backspace_1) - 1
```
Comment (AlekseiAfanasev) [inline homework7/hw2.py:47]: len(str_without_backspace_1)  may be omitted

File: homework7/hw3.py [hunk 6]
```diff
@@ -0,0 +1,69 @@
...
+    If board is unfinished, function should return "unfinished!"
+    param:
+        board (List[List]): input bord
+    return: result str
+    """
+    for line in board:
+        if len(set(line)) == 1:
+            return f"{line[0][0]} wins!"
+
+    for num in range(len(board)):
```
Comment (AlekseiAfanasev) [inline homework7/hw3.py:42]: I guess it should be not "num", but "column" or smth like that

File: homework7/tests/test_hw1.py [hunk 7]
```diff
@@ -0,0 +1,23 @@
+from homework7.hw1 import find_occurrences
+
+
+def test_find_str_in_dict_tree():
```
Comment (AlekseiAfanasev) [inline homework7/tests/test_hw1.py:4]: add more test cases. what if tree is empty? what if there is no occurences? what if occurency is in key? etc.

File: homework7/tests/test_hw2.py [hunk 8]
```diff
@@ -0,0 +1,15 @@
...
+
+
+def test_two_str_with_same_output_str():
+    """Test that function return True if we "#" symbol like backspace"""
+    str_1 = "we#c"
+    str_2 = "wetr###c"
+    assert backspace_compare(str_1, str_2)
+
+
+def test_two_str_with_not_the_same_output_str():
```
Comment (AlekseiAfanasev) [inline homework7/tests/test_hw2.py:11]: add more test cases. what if strings start with #? what if one has # another not? etc.

File: homework7/hw1.py (hunk 3 above)
Comment (Yury3077) [reply homework7/hw1.py:48]: Don't have any custom iterable types, bacause of task discription : Tree can only contains basic structures like:
str, list, tuple, dict, set, int, bool. But i think it's smarte to check  __iter__  object. Right?

**Pull request merged**